searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/evaluation/StringValue.java

```diff
@@ -31,10 +35,15 @@ public StringValue(String value) {
     @Override
     public TensorType type() { return TensorType.empty; }
 
-    /** Returns the hashcode of this, to enable strings to be encoded (with reasonable safely) as doubles for optimization */
+    /**
+     * Returns the XXHash hashcode of this, to enable strings to be encoded (with reasonable safely)
+     * as doubles for optimization.
+     */
     @Override
     public double asDouble() {
-        return value.hashCode();
+        XXHash64 hasher = XXHashFactory.fastestInstance().hash64();
```

Comment:
Does the hash function need to be the same in both java and c++. We are using the faster XXH3_64 as default for all hashing in the backend.

Reply:
I don't think we strictly depend on that any more, but it would feel safer to keep them the same ...

Reply:
We have no tests verifying equal hashing so I think we are fine there.
The backend have standardized on XXH3, and are using that for the string repo used for tensors and various other use cases. We probably want to use this for string attributes too in the not too distant future. So it is better that we just break it now. Once XXH3 is available for java too we can start using it there too.

Reply:
XXH3 is available not just i the jpountz implementation ... trying it now.

Reply:
I guess we should see where we are using jpountz library and unify usage - later.

Reply:
Yes, file distribution I think.